tests/test_editable.py

```diff
@@ -105,3 +106,50 @@ def test_cython_pxd(monkeypatch, tmp_path, editable, editable_mode, isolated):
         *editable_flag,
         ".",
     )
+
+
+@pytest.mark.compile()
+@pytest.mark.configure()
+@pytest.mark.integration()
+@pytest.mark.usefixtures("package_simplest_c")
+def test_install_dir(isolated):
+    isolated.install("pip>=23")
+    isolated.install("scikit-build-core")
+
+    settings_overrides = {
+        "build-dir": "build/{wheel_tag}",
+        "wheel.install-dir": "sub_pkg",
+        "editable.rebuild": "true",
+    }
+    # Create a dummy _module to satisfy the import
+    Path("./src/simplest/_module.py").write_text(
+        textwrap.dedent(
+            """
+            def square(__x: float, __y: float) -> float:
+                pass
+            """
+        )
+    )
+
+    isolated.install(
+        "-v",
+        *[f"--config-settings={k}={v}" for k, v in settings_overrides.items()],
+        "--no-build-isolation",
+        "-e",
+        ".",
+    )
+
+    # Make sure the package is correctly installed in the subdirectory
+    sub_pkg_path = isolated.platlib / "sub_pkg"
+    c_module_glob = list(sub_pkg_path.glob("simplest/_module*"))
+    assert len(c_module_glob) == 1
+    c_module = c_module_glob[0]
+    assert c_module.exists()
+    # If `install-dir` was not taken into account it would install here
+    failed_c_module = sub_pkg_path / "../simplest" / c_module.name
+    assert not failed_c_module.exists()
+
+    # Run an import in order to re-trigger the rebuild and check paths again
+    isolated.execute("import simplest")
```

Comment:
I've added some relevant test for this. Am I missing something here? Do I need to touch the CMake files or something to make sure the `import` triggers the `cmake --install`? Last time I remembered the rebuild was done at each `import` regardless right?

Reply:
Ok, I've made it import the C module to make sure `known_wheel_files` is triggered. I have cherry-picked the test onto main, and confirmed it fails at the `failed_c_module.exists()` part, so at least it seems to catch this specific case as intended. @henryiii do you want to check it as well?

Reply:
LGTM!